### DISCUSSION TITLE
User Settings?

### DISCUSSION THREAD

**Discussion by Spacefreak18** [mfussenegger/nvim-dap#383] [Q&A]
Is there a way I can specify user settings like here:
https://github.com/Microsoft/vscode-java-debug#Options

I see hotcodereplace , but i'm not sure it is the same things as I'm thinking.

Specifically I want java.debug.settings.showToString to be false, because I'm having a lot of lag when debugging certain paths, and that speeds things up in IntelliJ.

**Comment by Spacefreak18** [comment]
I changed some of the settings i use to start eclipse.jdt.ls and i deleted the eclipse.jdt.ls workspace folder and it was so fast!

I still think it's worth finding out if changing settings like the one i mentioned in my previous post is possible. I will report back though if deleting the workspace folder works going forward.

And perhaps if this is a better question for the eclipse.jdt.ls project, let me know.

**Reply by mfussenegger** [reply]
Most of these options are client options for the vscode-java-debug extension. 
For example the `showToString` value is used in a custom request: https://github.com/microsoft/vscode-java-debug/blob/6a56a5a9daf9978491f4895e1644ab05c1601092/src/variableMenu.ts#L49 (not part of the debug adapter protocol)

nvim-dap only supports the core protocol. Custom protocol extensions would have to be implemented as additional plugins.
